modules/task/execution.go

```diff
@@ -111,11 +111,15 @@ func Register(fn interface{}) error {
 	if ok {
 		return nil
 	}
-	// Register function types for encoding
 	for i := 0; i < fnType.NumIn(); i++ {
-		arg := reflect.Zero(fnType.In(i)).Interface()
-		if err := GlobalBackend().Encoder().Register(arg); err != nil {
-			return errors.Wrap(err, "unable to register the message for encoding")
+		argType := fnType.In(i)
+		// Interfaces cannot be registered, their implementations should be
+		// https://golang.org/pkg/encoding/gob/#Register
+		if argType.Kind() != reflect.Interface {
```

Comment:
First argument is context. This is only calling gob register if the type is not an interface. It does not do anything beyond that.